### DISCUSSION TITLE
Could not load file or assembly 'System.Data.DataSetExtensions'

### DISCUSSION THREAD

**Discussion by jaknoke** [arkypita/LaserGRBL#1222] [Q&A]
LaserGrbl v3.8.6.0
 v
Wrapper: UsbSerial
Win 7, Ultimate, 6.1.7601.65536, Service Pack 1 (6)
CLR: C:\windows\mono\mono-2.0\lib\mono\4.5\mscorlib.dll 4.6.57.0

TypeOf exception  [System.TypeInitializationException]
Exception message [The type initializer for 'LaserGRBL.UsageStats' threw an exception.]
Exception source  [LaserGRBL], thread []
Exception method  [Void Main(System.String[])]

   ----------- stack trace -----------
  at LaserGRBL.Program.Main (System.String[] args) [0x00015] in <4b3130430cd14d049ed6acee3cb3d9d1>:0 

Inner exception data
TypeOf exception  [System.TypeInitializationException]
Exception message [The type initializer for 'LaserGRBL.GrblCore' threw an exception.]
Exception source  [LaserGRBL], thread []
Exception method  [System.String get_filename()]

   ----------- stack trace -----------
  at LaserGRBL.Logger.get_filename () [0x00000] in <4b3130430cd14d049ed6acee3cb3d9d1>:0 
  at LaserGRBL.Logger.DoTheWork () [0x00030] in <4b3130430cd14d049ed6acee3cb3d9d1>:0 

Inner exception data
TypeOf exception  [System.TypeLoadException]
Exception message [Could not load type of field 'LaserGRBL.PSHelper.MaterialDB:FromServer' (0) due to: Could not load file or assembly 'System.Data.DataSetExtensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies. assembly:System.Data.DataSetExtensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:<unknown type> member:<none>]
Exception source  [], thread []
Exception method  []

**Comment by arkypita** [comment]
I suggest you to install or re-install Microsoft .NET framework

**Comment by arkypita** [comment]
Moved to [discussion](https://github.com/arkypita/LaserGRBL/discussions)

**Comment by arkypita** [comment]
> I suggest you to install or re-install Microsoft .NET framework

Does it fix?

I read : CLR: C:\windows\mono\mono-2.0\lib\mono\4.5\mscorlib.dll 4.6.57.0
Are you using it on Linux?

**Comment by jaknoke** [comment]
It did fix the issue, however when I try to load an image, the program
freezes.  It shows the loading bar at the bottom right of the screen.  I
waited for over an hour and a half and the image never loaded.
